lasagne/layers/special.py

```diff
@@ -452,23 +454,37 @@ def _transform_affine(theta, input, downsample_factor):
     input_dim = input.dimshuffle(0, 2, 3, 1)
     input_transformed = _interpolate(
         input_dim, x_s_flat, y_s_flat,
-        out_height, out_width)
+        out_height, out_width, border_mode)
 
     output = T.reshape(
         input_transformed, (num_batch, out_height, out_width, num_channels))
     output = output.dimshuffle(0, 3, 1, 2)  # dimshuffle to conv format
     return output
 
 
-def _interpolate(im, x, y, out_height, out_width):
+def _interpolate(im, x, y, out_height, out_width, border_mode):
     # *_f are floats
     num_batch, height, width, channels = im.shape
     height_f = T.cast(height, theano.config.floatX)
     width_f = T.cast(width, theano.config.floatX)
 
     # clip coordinates to [-1, 1]
-    x = T.clip(x, -1, 1)
-    y = T.clip(y, -1, 1)
+    if border_mode == 'nearest':
+        x = T.clip(x, -1, 1)
+        y = T.clip(y, -1, 1)
+    # 0.9 1.0 1.1 -> 0.9 1.0 0.9
+    elif border_mode == 'reflect':
+        xa = T.maximum(x, -(2 + x))
+        ya = T.maximum(y, -(2 + y))
+        x = T.minimum(xa, 2 - xa)
+        y = T.minimum(ya, 2 - ya)
```

Comment:
Almost! It needs to be periodic just like `wrap`. E.g., when you go further, 2.9 should be mapped to -.9 (which works), 3.0 should be mapped to -1 (which works), and 3.1 should be mapped to -.9 again (mapped to -1.1 now).